service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcDocumentTaskExecutor.java

```diff
@@ -46,9 +53,27 @@ public void execute(GetGpcDocumentTaskDefinition taskDefinition) {
         var storageDataWrapper = StorageDataWrapperProvider.buildStorageDataWrapper(taskDefinition, response, taskId);
         storageConnectorService.uploadFile(storageDataWrapper, documentName);
 
+        String messageId = randomIdGeneratorService.createNewId();
+
         EhrExtractStatus ehrExtractStatus = ehrExtractStatusService.updateEhrExtractStatusAccessDocument(taskDefinition, documentName,
-            taskId);
+            taskId, messageId);
+
+        String xmlPayload = translateToXmlPayload(taskDefinition, messageId);
+        var storageDataWrapperWithMhsPayload = StorageDataWrapperProvider.buildStorageDataWrapper(taskDefinition, xmlPayload, taskId);
+        storageConnectorService.uploadFile(storageDataWrapperWithMhsPayload, prepareXmlDocumentName(messageId));
```

Comment:
I think we should updated EhrExtract status with the new file name that was just uploaded.